### ISSUE TITLE
aria-haspopup with value treegrid?

### ISSUE THREAD

**Issue by JAWS-test** [w3c/aria#1333]
aria-haspopup can have the values grid and tree. So it is obvious that treegrid should be allowed too, right?

**Comment by stes-acc** [comment]
+1

We should also clarify what value aria-haspopup should have if container has role dialog but the only instance inside is a tree, a menu etc.

These things HAPPEN because developers think of popops being MODAL/NON-MODAL and all available (APG) examples available don't even mention modality of roles tree, menu and so on just saying menu ec. is root container.

The caveat with this is that dialog derives from **window** whereas menu etc. derives from **widget**. Window has clearly defined modal/non-modal properties whereas for widget this is just not defined, which causes confusion among devs.

So two questions for clarification:

1. **The ARIA spec is lacking a sentence like "menu role used as container is always modal/non-modal" (pick your preference)**

2. Alternatively, is is OK to "wrap" such popups in dialogs with aria-modal="true/false" and give aria-haspopup the value of the inner content IF THIS IS THE ONLY CONTENT IN THE DIALOG?

**Comment by scottaohara** [comment]
making a note here that further attribute clarification discussion, beyond the original question of this thread, should be tracked in #1024, so that we don't have duplicate threads going on.
